lib/src/adapters/persistent_json_api.dart

```diff
@@ -172,9 +177,15 @@ class PersistentJsonApiAdapter extends JsonApiAdapter {
     return id;
   }
 
-  Future<Iterable<JsonApiDocument?>> addedByEndpoint(String endpoint) async {
+  Future<Iterable<JsonApiDocument>> addedByEndpoint(String endpoint) async {
     var box = await openBox('added');
-    return box.values.where((doc) => doc!.endpoint == endpoint);
+    List<JsonApiDocument> docs = [];
+    box.values.forEach((doc) {
```

Comment:
Similar casting instead of `forEach` could be performed here. I'm not against good old iterations, but as for me, functional list transformations look more concise.